src/mainwindow.cpp

```diff
@@ -731,19 +743,56 @@ void MainWindow::onOpenMapListContextMenu(const QPoint &point)
 void MainWindow::onAddNewMapToGroupClick(QAction* triggeredAction)
 {
     int groupNum = triggeredAction->data().toInt();
-    QStandardItem* groupItem = mapGroupItemsList->at(groupNum);
+    openNewMapPopupWindow(groupNum);
+}
+
+void MainWindow::onNewMapCreated() {
+    QString newMapName = this->newmapprompt->map->name;
+    int newMapGroup = this->newmapprompt->group;
+    Map *newMap_ = this->newmapprompt->map;
+
+    Map *newMap = editor->project->addNewMapToGroup(newMapName, newMapGroup, newMap_);
+
+    qDebug() << "Created a new map named" << newMapName;
 
-    QString newMapName = editor->project->getNewMapName();
-    Map* newMap = editor->project->addNewMapToGroup(newMapName, groupNum);
     editor->project->saveMap(newMap);
     editor->project->saveAllDataStructures();
 
+    QStandardItem* groupItem = mapGroupItemsList->at(newMapGroup);
     int numMapsInGroup = groupItem->rowCount();
-    QStandardItem *newMapItem = createMapItem(newMapName, groupNum, numMapsInGroup);
+
+    QStandardItem *newMapItem = createMapItem(newMapName, newMapGroup, numMapsInGroup);
     groupItem->appendRow(newMapItem);
     mapListIndexes.insert(newMapName, newMapItem->index());
 
-    setMap(newMapName);
+    setMap(newMapName, true);
+
+    if (newMap->isFlyable == "TRUE") {
+        addNewEvent("event_heal_location");
+        editor->project->saveHealLocationStruct(newMap);
```

Comment:
Ideally we'd just save one time in this function, but this is doing an extra save just for the heal location.  an saveAllDataStructures() be move down here, or was there a complication with that?

Reply:
Well it was a few months ago but I'm 70% sure there was a reason for this -- I will try to determine.

Reply:
Oh yes, it was because I couldn't save the event with existing functions (`addNewEvent`) while the map was not set in the editor, and I had to save before setting the map. I suppose I could try to change this if you wnat?

Reply:
it's fine for now.  was just wondering if it could easily be removed here.